realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -476,8 +484,70 @@ ObjectServerUser getSyncUser() {
         return syncUser;
     }
 
+    // Class wrapping requests made against the auth server. Is also responsible for calling with success/error on the
+    // correct thread.
+    private static abstract class Request {
+
+        private final Callback callback;
+        private final AndroidRealmNotifier handler;
+        private final ThreadPoolExecutor networkPoolExecutor;
+
+        public Request(ThreadPoolExecutor networkPoolExecutor, Callback callback) {
+            this.callback = callback;
+            this.handler = new AndroidRealmNotifier(null, new AndroidCapabilities());
+            this.networkPoolExecutor = networkPoolExecutor;
+        }
+
+        // Implements the request. Return the current sync user if the request succeeded. Otherwise throw an error.
+        public abstract SyncUser run() throws ObjectServerError;
+
+        // Start the request
+        public RealmAsyncTask start() {
+            Future<?> authenticateRequest = networkPoolExecutor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        postSuccess(Request.this.run());
+                    } catch (ObjectServerError e) {
+                        postError(e);
+                    }
+                }
+            });
+            return new RealmAsyncTaskImpl(authenticateRequest, networkPoolExecutor);
+        }
+
+        private void postError(final ObjectServerError error) {
+            if (callback != null) {
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            callback.onError(error);
+                        } catch (Exception e) {
+                            RealmLog.error(e, "onError has thrown an exception that is being ignored.");
+                        }
+                    }
+                });
+            } else {
+                RealmLog.error(error, "An error was thrown, but no callback was available");
```

Comment:
I don't think logging `error` that way is a good idea. Since some people do monitor the log, and it would be a false positive. I changed the logic slightly instead so we should now log in all cases where something else doesn't handle it.